proxy.go

```diff
@@ -102,6 +102,21 @@ func NewProxy(config Config) *Proxy {
 	return proxy
 }
 
+// SetTimeout set Timeout of the proxy.
+func (p *Proxy) SetTimeout(timeout time.Duration) {
+	p.timeout = timeout
+}
+
+// SetTransport set RoundTripper of the proxy.
+func (p *Proxy) SetTransport(transport *http.Transport) {
```

Comment:
Could you please explain what's the need for exposing transport?

Reply:
The need is to be able to control the transport configuration (dialer, proxy, etc.). I need to modify the DNS resolver to avoid service discovery in a k8s cluster.

Reply:
I was just thinking that it'd be better to route `transport` through the `proxy.dial`, the current approach when the transport is independent looks pretty strange.

It will also effectively solve this issue: https://github.com/AdguardTeam/gomitmproxy/issues/24 without any config changes.

Reply:
like this https://github.com/google/martian/blob/0f7e6797a04da412118541344bbe0d65945e24c9/proxy.go#L129 ?